### PR TITLE
feat(query): support parse create dictionary stmt

### PR DESCRIPTION
**Pull request by Winnie-Hong0927**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

support parse create/drop/show dictionary stmt

part of #15901

## Tests

- [x] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16137)

### CODE REVIEW

File: src/query/ast/src/ast/format/ast_format.rs [hunk 1]
```diff
@@ -1749,6 +1771,93 @@ impl<'ast> Visitor<'ast> for AstFormatVisitor {
...
+        let comment_node = FormatTreeNode::new(comment_format_ctx);
+        children.push(comment_node);
+        let name = "CreateDictionary".to_string();
+        let format_ctx = AstFormatContext::with_children(name, children.len());
+        let node = FormatTreeNode::with_children(format_ctx, children);
+        self.children.push(node);
+    }
+
+    fn visit_drop_dictionary(&mut self, stmt: &'ast DropDictionaryStmt) {
+        self.visit_identifier(&stmt.dictionary_name);
```
Comment (b41sh) [inline src/query/ast/src/ast/format/ast_format.rs:1840]: use `visit_dictionary_ref` here.

File: src/query/ast/src/ast/format/ast_format.rs [hunk 2]
```diff
@@ -1749,6 +1771,93 @@ impl<'ast> Visitor<'ast> for AstFormatVisitor {
...
+        let child = self.children.pop().unwrap();
+        let name = "DropDictionary".to_string();
+        let format_ctx = AstFormatContext::with_children(name, 1);
+        let node = FormatTreeNode::with_children(format_ctx, vec![child]);
+        self.children.push(node);
+    }
+
+    fn visit_show_create_dictionary(&mut self, stmt: &'ast ShowCreateDictionaryStmt) {
+        self.visit_identifier(&stmt.dictionary_name);
```
Comment (b41sh) [inline src/query/ast/src/ast/format/ast_format.rs:1849]: ditto

**Comment by BohuTANG** [comment]
@Winnie-Hong0927 

Thanks for the contribution!
Can you open a new issue for tracking the dictonary feature? For example: https://github.com/datafuselabs/databend/issues/16135

**Comment by b41sh** [comment]
> @Winnie-Hong0927
> 
> Thanks for the contribution! Can you open a new issue for tracking the dictonary feature? For example: #16135

we already have a tracking issue #15901

**Review by b41sh — APPROVED** [review]
LGTM, thanks for your contribution @Winnie-Hong0927

**Review by sundy-li — APPROVED** [review]

**Pull request merged**